homeassistant/components/homekit/__init__.py

```diff
@@ -0,0 +1,143 @@
+"""Support for Apple Homekit.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/homekit/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from pyhap.accessory_driver import AccessoryDriver
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_NAME, CONF_IP_ADDRESS, CONF_PORT,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+from .accessories import HomeBridge
+from .covers import Window
+from .sensors import TemperatureSensor
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'homekit'
+REQUIREMENTS = ['HAP-python==1.1.2']
+
+ATTR_CLASS = 'class'
+ATTR_NAME = 'name'
+
+CONF_PIN_CODE = 'pincode'
+CONF_TYPES = 'types'
+
+ALL_TYPES = {
+    'sensor_temperature': {ATTR_NAME: 'Temperature',
+                           ATTR_CLASS: TemperatureSensor},
+    'cover': {ATTR_NAME: 'Cover',
+              ATTR_CLASS: Window},
+}
+
+
+def valid_typ(value):
+    """Check if typ is valid."""
+    if value.lower() in ALL_TYPES.keys():
+        return value.lower()
+    raise vol.Invalid('Typ {} is an invalid typ.'.format(value))
+
+
+def valid_string(string):
+    """Check if name string is valid. An empty string is allowed."""
+    if string is None:
+        return None
+    return str(string)
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All({
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_PORT): vol.Coerce(int),
+        vol.Optional(CONF_PIN_CODE): cv.string,
+        vol.Required(CONF_TYPES): vol.Schema({
```

Comment:
It doesn't make much sense to have the user define that a cover entity needs to be exposed as a cover in homekit. We should be able to figure that out ourselves.

By default we should expose everything we know and allow the user to configure a filter, just like we do for Alexa or Google Home. [Alexa example code](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/alexa/__init__.py#L33-L36)

Reply:
For the initial PR, I think it is totally fine to leave filtering out and just expose all covers and temp sensors by name as they are known in Home Assistant.

Reply:
Add this to async_setup to just loop through all entities:

```python
for state in hass.states.async_all():
    if state.domain == 'cover':
        # Register cover entity
    elif … # etc
```